src/RangeBase.ts

```diff
@@ -19,184 +23,256 @@ interface RangeBaseConfig extends BaseletConfig {
+            Object.prototype.hasOwnProperty.call(data, idKey)
+          )
+        ) {
+          return Promise.reject(
+            new Error(`data must have properties ${rangeKey} and ${idKey}`)
+          )
+        }
+        const bucketNumber = Math.floor(data[rangeKey] / bucketSize)
+        const bucketFilename = `${bucketNumber}.json`
+        const bucketPath = `${databaseName}${formattedPartition}/${bucketFilename}`
+
+        return disklet
+          .getText(bucketPath)
+          .then(serializedBucket => JSON.parse(serializedBucket))
+          .then(
+            bucketData => {
+              const firstRangeOccurence = getIndex(
+                data[rangeKey],
+                rangeKey,
+                bucketData
+              )
+              if (firstRangeOccurence.found === false) {
+                bucketData.splice(firstRangeOccurence.index, 0, data)
+              } else {
+                const lastRangeOccurence = getIndex(
+                  data[rangeKey],
+                  rangeKey,
+                  bucketData,
+                  firstRangeOccurence.index,
+                  bucketData.length - 1,
+                  true
+                )
+                const targetIndex = getIndex(
+                  data[idKey],
+                  idKey,
+                  bucketData,
+                  firstRangeOccurence.index,
+                  lastRangeOccurence.index
+                )
+                if (targetIndex.found === true) {
+                  throw new Error(
+                    'Cannot insert data because id already exists'
+                  )
+                }
+                bucketData.splice(targetIndex.index, 0, data)
+              }
+              return disklet.setText(bucketPath, JSON.stringify(bucketData))
+            },
+            error => {
+              console.log(error)
+              console.log('assuming bucket doesnt exist')
+              // we must be sure bucket doesnt exist, else we'll overwrite
+              // we have to identify the error
+              return disklet.setText(bucketPath, JSON.stringify([data]))
+            }
+          )
+      },
+      query(
+        partition: string = '/',
+        rangeStart: number,
+        rangeEnd: number = rangeStart
+      ): Promise<any[]> {
+        const { bucketSize, rangeKey } = configData
+        const formattedPartition = checkAndformatPartition(partition)
+        // sanity check the range
+        const bucketFetchers = []
+        for (
+          let bucketNumber = Math.floor(rangeStart / bucketSize);
+          bucketNumber <= Math.floor(rangeEnd / bucketSize);
+          bucketNumber++
+        ) {
+          bucketFetchers.push(
+            disklet
+              .getText(
+                `${databaseName}${formattedPartition}/${bucketNumber}.json`
+              )
+              .then(rawBucketData => JSON.parse(rawBucketData))
+              .catch(error => {
+                console.log(
```

Comment:
Only Console.log if there was an error parsing the data. If the file does not exist, that is OK and doesn't deserve any logging